File: FournisseurIdentite/utils/ServiceEmail.cs
using MailKit.Net.Smtp;
using MimeKit;
using System.Threading.Tasks;
namespace FournisseurIdentite.Utils
{
    public class ServiceEmail
    {
        private readonly string _smtpServer = "smtp.gmail.com"; // Remplacez par votre serveur SMTP
        private readonly int _smtpPort = 587; // Port SMTP (souvent 587 ou 465)
        private readonly string _smtpUsername = "Zotahiana691@gmail.com"; // Remplacez par votre adresse email
        private readonly string _smtpPassword = "mnyb jket tdbp anvv"; // Remplacez par votre mot de passe

        public async Task EnvoyerAsync(string toEmail, string sujet, string message)
        {
            var emailMessage = new MimeMessage();
            emailMessage.From.Add(new MailboxAddress("Nom de votre entreprise", _smtpUsername));
            emailMessage.To.Add(new MailboxAddress("", toEmail));
            emailMessage.Subject = sujet;

            var body = new TextPart("plain")
            {
                Text = message
            };

            emailMessage.Body = body;

            using (var client = new SmtpClient())
            {
                try
                {
                    // Connexion au serveur SMTP
                    await client.ConnectAsync(_smtpServer, _smtpPort, false);

                    // Authentification avec le serveur SMTP
                    await client.AuthenticateAsync(_smtpUsername, _smtpPassword);

                    // Envoi de l'email
                    await client.SendAsync(emailMessage);
                }
                catch (Exception ex)
                {
                    // Gérer l'exception, loggez ou re-throw selon le cas
                    Console.WriteLine($"Erreur lors de l'envoi de l'email: {ex.Message}");
                    throw;
                }
                finally
                {
                    // Déconnexion propre du client
                    await client.DisconnectAsync(true);
                    client.Dispose();
                }
            }
        }
    }
}

File: crypto/serveur/src/main/java/mg/itu/crypto/utils/ServiceEmail.java
package mg.itu.crypto.utils;

import jakarta.mail.*;
import jakarta.mail.internet.InternetAddress;
import jakarta.mail.internet.MimeMessage;
import java.util.Properties;

public class ServiceEmail {

    private final String smtpServer = "smtp.gmail.com";
    private final int smtpPort = 587;
    private final String smtpUsername = "Zotahiana691@gmail.com";
    private final String smtpPassword = "mnyb jket tdbp anvv"; // À sécuriser !

    public void envoyer(String toEmail, String sujet, String message) {
        // Propriétés de configuration SMTP
        Properties properties = new Properties();
        properties.put("mail.smtp.auth", "true");
        properties.put("mail.smtp.starttls.enable", "true");
        properties.put("mail.smtp.host", smtpServer);
        properties.put("mail.smtp.port", String.valueOf(smtpPort));

        // Création de la session avec authentification
        Session session = Session.getInstance(properties, new Authenticator() {
            @Override
            protected PasswordAuthentication getPasswordAuthentication() {
                return new PasswordAuthentication(smtpUsername, smtpPassword);
            }
        });

        try {
            // Création du message
            Message emailMessage = new MimeMessage(session);
            emailMessage.setFrom(new InternetAddress(smtpUsername, "Nom de votre entreprise"));
            emailMessage.setRecipients(Message.RecipientType.TO, InternetAddress.parse(toEmail));
            emailMessage.setSubject(sujet);
            emailMessage.setText(message);

            // Envoi du message
            Transport.send(emailMessage);
            System.out.println("Email envoyé avec succès à " + toEmail);
        } catch (Exception e) {
            System.err.println("Erreur lors de l'envoi de l'email : " + e.getMessage());
        }
    }
}
